compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1176,6 +1176,19 @@ public void loadRawTemplateType() {
         throw new IllegalStateException("'lang.object:RawTemplate' type not found");
     }
 
+    public void loadNaturalGeneratorType() {
+        ScopeEntry entry = symTable.langNaturalModuleSymbol.scope.lookup(Names.NATURAL_GENERATOR);
+        while (entry != NOT_FOUND_ENTRY) {
+            if ((entry.symbol.tag & SymTag.TYPE) != SymTag.TYPE) {
+                entry = entry.next;
+                continue;
+            }
+            symTable.naturalGeneratorType = (BObjectType) entry.symbol.type;
```

Comment:
Don't we need `instanceof BObjectType` check here.

Reply:
Since this is a langlib construct, IMO, we don't need the check.